src/eos-updater-poll-lan.c

```diff
@@ -0,0 +1,920 @@
+        swbf1->branch_file->download_time;
+      GDateTime *timestamp2 = swbf2->declared_download_time ?
+        swbf2->declared_download_time :
+        swbf2->branch_file->download_time;
+
+      return g_date_time_compare (timestamp2, timestamp1);
+    }
+  // one has signature, the other not
+  if (with_signature1)
+    return -1;
+  return 1;
+}
+
+static gboolean
+filter_services (LanData *lan_data,
+                 GPtrArray *found_services,
+                 GPtrArray **out_valid_services,
+                 GError **error)
+{
+  guint idx;
+  g_autoptr(GPtrArray) valid_services = object_array_new ();
+
+  for (idx = 0; idx < found_services->len; ++idx)
+    {
+      gpointer service_ptr = g_ptr_array_index (found_services, idx);
+      EosAvahiService *service = EOS_AVAHI_SERVICE (service_ptr);
+      TxtRecordError txt_error = TXT_RECORD_OK;
+      const gchar *txt_version = get_unique_txt_record (service->txt,
+                                                        "eos_txt_version",
+                                                        &txt_error);
+      ServiceCheck checker;
+      gboolean valid = FALSE;
+      g_autoptr(EosServiceWithBranchFile) swbf = NULL;
+
+      if (txt_version == NULL)
+        {
+          // TODO: message
+          continue;
+        }
+      checker = get_txt_handler (txt_version);
+      if (checker == NULL)
+        {
+          // TODO: message
+          continue;
+        }
+      swbf = eos_service_with_branch_file_new (service);
+      if (!checker (lan_data, swbf, &valid, error))
+        return FALSE;
+
+      if (!valid)
+        continue;
+      g_ptr_array_add (valid_services, g_steal_pointer (&swbf));
+    }
+
+  g_ptr_array_sort (valid_services, g_compare_func_swbf_by_timestamp);
+  *out_valid_services = g_steal_pointer (&valid_services);
+  return TRUE;
+}
+
+static gchar *
+get_branch_file_uri (EosAvahiService *service)
+{
+  return g_strdup_printf ("http://%s:%" G_GUINT16_FORMAT "/extensions/eos/branch_file",
+                          service->address,
+                          service->port);
+}
+
+static EosBranchFile *
+get_newest_branch_file (LanData *lan_data,
+                        GPtrArray *valid_services)
+{
+  guint idx;
+  GCancellable *cancellable;
+
+  cancellable = g_task_get_cancellable (lan_data->fetch_data->task);
+  for (idx = 0; idx < valid_services->len; ++idx)
+    {
+      gpointer swbf_ptr = g_ptr_array_index (valid_services, idx);
+      EosServiceWithBranchFile* swbf = EOS_SERVICE_WITH_BRANCH_FILE (swbf_ptr);
+      g_autoptr(GBytes) branch_file_contents = NULL;
+      g_autoptr(GBytes) signature_contents = NULL;
+      g_autofree gchar *uri = get_branch_file_uri (swbf->service);
+      g_autoptr(GError) error = NULL;
+      g_autoptr(EosBranchFile) branch_file = NULL;
+
+      if (swbf->branch_file->branch_file != NULL)
+        return g_object_ref (swbf->branch_file);
+
+      uri = get_branch_file_uri (swbf->service);
```

Comment:
It is fixed in one of those "review fixes" commits. But github insist on showing this as an non-outdated diff for whatever reason.